dev_guide/expanding_persistent_volumes.adoc

```diff
@@ -0,0 +1,89 @@
+[[expanding_persistent_volumes]]
+= Expanding Persistent Volumes
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+[[enabling_expansion_pvc]]
+== Enabling Expansion of Persistent Volume Claims
+
+Volume expansion is a Technology Preview feature, and hence, is not enabled by
+default in your {product-title} {product-version} cluster. There may be other reasons that
+{product-title} administrators wish to enable this feature for certain use
+cases.
+
+[NOTE]
+====
+For more information on Red Hat Technology Preview features support scope,  see
+https://access.redhat.com/support/offerings/techpreview/. 
+====
+
+To allow expansion of persistent volume claims (PVC) by {product-title} users,
+{product-title} administrators must create or update a StorageClass with
+`allowVolumeExpansion` set to `true`. Only PVCs created from that class are
+allowed to expand.
+
+Apart from that, {product-title} administrators must enable the
+`ExpandPersistentVolumes` feature flag and turn on the
+`PersistentVolumeClaimResize` admission controller. Refer to <link_here>
+for how to enable a feature flag and how to enable an admission controller.
```

Comment:
This still needs fixing. @vikram-redhat do we document feature flag stuff anywhere in Openshift? I know that, Openshift-ansible supports it.

Reply:
@gnufied - admission controllers are described here: https://docs.openshift.com/container-platform/3.7/architecture/additional_concepts/admission_controllers.html. Is that correct?

Reply:
@tmorriso-rh - where were you thinking of pointing the user when you wrote this?

Reply:
@vikram-redhat This was not part of the original PR. @gnufied suggested this change. I have asked for the correct link.

Reply:
To allow expansion of PVCs:
1st. Need to enable feature gate: ExpandPersistentVolumes
2nd. Better to enable admission controller: PersistentVolumeClaimResize, cause enable this admission controller can control which PVCs can be allowed expansion.
3rd. If enabled admission controller, only the PVCs created from a storageclass with "allowVolumeExpansion" is set to "true"

Reply:
Thanks @qinpingli. @gnufied and @childsb Please see the comment above from QE.

Reply:
@vikram-redhat @tmorriso-rh yes, I suggested that because most tech. preview features also document how to enable them. If our users can't figure out, how to enable particular feature then it would be hard for them to use it. 


@vikram-redhat yep that link for admission controller is correct. We should use it in this document.  @tmorriso-rh I could not find an existing link for - how to enable an alpha feature. But https://github.com/openshift/openshift-docs/blob/master/scaling_performance/managing_hugepages.adoc page for example talks about it. We should have similar information in this document about, how to enable `ExpandPersistentVolumes` feature flag in master and node-config. Please update this document to reflect that.

Reply:
@gnufied - we do document how to enable tech preview features here: https://docs.openshift.com/container-platform/3.7/install_config/web_console_customization.html#web-console-enable-tech-preview-feature. Would this be a better link or an additional link (admittedly, the user has to know the name of the feature that they want enabling)?

Reply:
@vikram-redhat that looks like web console customization whereas this feature flag has to be turned on explicitly in node and master configs.  We need to document, how to enable resizing feature flags.

Reply:
@gnufied Ah ok. I don't know that myself. @tmorriso-rh @ahardin-rh please note @gnufied comment.

Reply:
Please replace "Refer to <link_here> for how to enable a feature flag and how to enable an admission controller." with:

Refer to https://docs.openshift.com/container-platform/3.7/architecture/additional_concepts/admission_controllers.html document for enabling PersistentVolumeClaimResize admission controller.  

To enable `ExpandPersistentVolumes`  feature. It must be set to `true` across the system.

1. node-config.yaml of all nodes in cluster must have:

```yaml
# cat /etc/origin/node/node-config.yaml
...
kubeletArguments:
...
  feature-gates:
  - ExpandPersistentVolumes=true
```

2. The master API also must have `ExpandPersistentVolumes` feature gate enabled:

```yaml
# cat /etc/origin/master/master-config.yaml
...
kubernetesMasterConfig:
  apiServerArguments:
  ...
  feature-gates:
  - ExpandPersistentVolumes=true

# systemctl restart atomic-openshift-master-api
```

Reply:
Should the link to Admission Controllers be for 3.7 or 3.9? This PR applies to 3.9.